File: Dummy/hawkshaw/library/datalayer/models/accessibility/SocialMediaKt.java
package com.hawkshaw.library.datalayer.models.accessibility;

import Y1.K;
import com.hawkshaw.library.datalayer.models.accessibility.PushSocialMediaRequest;
import com.hawkshaw.library.datalayer.models.accessibility.PushUnprocessedSocialMediaRequest;
import com.hawkshaw.library.datalayer.room.socialmedia.SocialMediaEntity;
import com.hawkshaw.library.datalayer.room.socialmedia.UnprocessedSocialMediaEntity;

public final class SocialMediaKt {
    public static final PushSocialMediaRequest.SocialMediaMessage toRequest(SocialMediaEntity socialMediaEntity) {
        K.n(socialMediaEntity, "<this>");
        return new PushSocialMediaRequest.SocialMediaMessage(socialMediaEntity.getCcn(), socialMediaEntity.getCcs(), socialMediaEntity.getSender(), socialMediaEntity.getMessage(), socialMediaEntity.getType(), socialMediaEntity.getTimestamp(), socialMediaEntity.getApplication(), socialMediaEntity.getTime());
    }

    public static final PushUnprocessedSocialMediaRequest.UnprocessedSocialMediaMessage toRequest(UnprocessedSocialMediaEntity unprocessedSocialMediaEntity) {
        K.n(unprocessedSocialMediaEntity, "<this>");
        return new PushUnprocessedSocialMediaRequest.UnprocessedSocialMediaMessage(unprocessedSocialMediaEntity.getId(), unprocessedSocialMediaEntity.getTitle(), unprocessedSocialMediaEntity.getStatus(), unprocessedSocialMediaEntity.getSender(), unprocessedSocialMediaEntity.getMessage(), unprocessedSocialMediaEntity.getType(), unprocessedSocialMediaEntity.getApplication(), unprocessedSocialMediaEntity.getTimeString());
    }
}

File: app/src/main/java/com/hawkshaw/library/datalayer/models/accessibility/SocialMediaKt.kt
package com.hawkshaw.library.datalayer.models.accessibility

import com.hawkshaw.library.datalayer.models.PackageName
import com.hawkshaw.library.datalayer.room.socialmedia.SocialMediaEntity
import com.hawkshaw.library.datalayer.room.socialmedia.UnprocessedSocialMediaEntity

/**
 * Extension function to convert a SocialMediaEntity to a PushSocialMediaRequest.SocialMediaMessage
 *
 * @return A new SocialMediaMessage instance with values from this SocialMediaEntity
 */
fun SocialMediaEntity.toRequest(): PushSocialMediaRequest.SocialMediaMessage {
    return PushSocialMediaRequest.SocialMediaMessage(
        ccn = this.packageName,
        sender = this.sender,
        message = this.message ?: "",
        // FIX: this.type is already SocialMediaEntity.Type, no need for valueOf()
        type = this.type,
        timestamp = this.timestamp,
        application = PackageName.from(this.packageName),
        time = this.timestamp.toString()
    )
}

/**
 * Extension function to convert an UnprocessedSocialMediaEntity to a PushUnprocessedSocialMediaRequest.UnprocessedSocialMediaMessage
 *
 * @return A new UnprocessedSocialMediaMessage instance with values from this UnprocessedSocialMediaEntity
 */
fun UnprocessedSocialMediaEntity.toRequest(): PushUnprocessedSocialMediaRequest.UnprocessedSocialMediaMessage {
    return PushUnprocessedSocialMediaRequest.UnprocessedSocialMediaMessage(
        id = this.id,
        title = this.title,
        status = this.status,
        sender = this.sender,
        message = this.message,
        type = this.type,
        application = this.application,
        timeString = this.timeString
    )
} 